pennylane/qnode_new/jacobian.py

```diff
@@ -0,0 +1,528 @@
+
+        if all(k in ('A', 'A2') for k in methods):
+            return 'A'
+
+        if None in methods:
+            return None
+
+        return 'F'
+
+    def jacobian(self, args, kwargs=None, *, wrt=None, method='B', options=None):
+        r"""Compute the Jacobian of the QNode.
+
+        Returns the Jacobian of the parametrized quantum circuit encapsulated in the QNode.
+        The Jacobian is returned as a two-dimensional array. The (possibly nested) input arguments
+        of the QNode are :func:`flattened <_flatten>` so the QNode can be interpreted as a simple
+        :math:`\mathbb{R}^m \to \mathbb{R}^n` function.
+
+        The Jacobian can be computed using several methods:
+
+        * Finite differences (``'F'``). The first order method evaluates the circuit at
+          :math:`n+1` points of the parameter space, the second order method at :math:`2n` points,
+          where ``n = len(wrt)``.
+
+        * Analytic method (``'A'``). Works for all one-parameter gates where the generator
+          only has two unique eigenvalues; this includes one-parameter single-qubit gates.
+          Additionally, can be used in CV systems for Gaussian circuits containing only first-
+          and second-order observables.
+          The circuit is evaluated twice for each incidence of each parameter in the circuit.
+
+        * Best known method for each parameter (``'B'``): uses the analytic method if
+          possible, otherwise finite difference.
+
+        * Device method (``'D'``): Delegates the computation of the Jacobian to the
+          device executing the circuit.
+
+        .. note::
+           The finite difference method is sensitive to statistical noise in the circuit output,
+           since it compares the output at two points infinitesimally close to each other. Hence the
+           'F' method requires exact expectation values, i.e., ``analytic=True`` in simulation plugins.
+
+        Args:
+            args (nested Iterable[float] or float): positional arguments to the quantum function (differentiable)
+            kwargs (dict[str, Any]): auxiliary arguments to the quantum function (not differentiable)
+            wrt (Sequence[int] or None): Indices of the flattened positional parameters with respect
+                to which to compute the Jacobian. None means all the parameters.
+                Note that you cannot compute the Jacobian with respect to the kwargs.
+            method (str): Jacobian computation method, in ``{'F', 'A', 'B', 'D'}``, see above
+            options (dict[str, Any]): additional options
+
+                * h (float): finite difference method step size
+                * order (int): finite difference method order, 1 or 2
+                * force_order2 (bool): force the analytic method to use the second-order method
+                * shots (int): number of times the circuit is evaluated (or sampled) to estimate expectation values FIXME is this used?
```

Comment:
If it's not being used, what *is* being used to estimate expectation values on hardware?

Reply:
Shots are definitely being used, however I'm not sure they are being used _here_. Instead, the device has its own `shots` attribute that is used. It can likely be removed from here.

Reply:
OK, I thought it might have been moved into Device.

Reply:
Removed.